src/kv/map.h

```diff
@@ -110,8 +108,25 @@ namespace kv
 
     LocalCommits empty_commits;
 
+    template <typename... Args>
+    LocalCommit* create_new_local_commit(Args&&... args)
+    {
+      LocalCommit* c = empty_commits.pop();
+      if (c == nullptr)
+      {
+        c = new LocalCommit(std::forward<Args>(args)...);
+      }
+      else
+      {
+        c->~LocalCommit();
+        new (c) LocalCommit(std::forward<Args>(args)...);
+      }
+      return c;
+    }
+
+  public:
```

Comment:
Big diff here is actually just "make constructor and clone public", so they can be called by the non-friend `Store`.